chainermn/dataset.py

```diff
@@ -15,6 +16,8 @@ def scatter_dataset(dataset, comm):
         dataset: A dataset (e.g., ``list``, ``numpy.ndarray``,
             ``chainer.datasets.TupleDataset``, ...).
         comm: ChainerMN communicator or MPI4py communicator.
+        shuffle: Shuffle the dataset before being scattered.
+        root: The root process of the scatter operation.
```

Comment:
* Could you add type specification to these two arguments? i.e., `shuffle (bool)` and `root (int)`. `dataset` and `comm` accepts various types, and thus we omitted the type specification (following the convention of Chainer, e.g., https://github.com/chainer/chainer/blob/master/chainer/iterators/serial_iterator.py#L26 )

* Add explanation of `seed`.